### PR TITLE
Signup: Add small_screen property for AB test

### PR DESCRIPTION
**Pull request by keoshi**
This PR adds a `small_screen` property for the mobile plans tables AB test. This will allow us to target just mobile devices to run the correct hypothesis. More details in `#comment-4559` : p6TEKc-1V1-p2

### Changes

- WPCOM: adds property to event `calypso_wp_plans_test_view`.
- JP: adds property to event `calypso_plans_view`.

### Returns

- The returned value of this function is `'1'` if screen is less than, or equal to, 660 pixels wide.
- Otherwise it returns `'0'`.

### References

Test name: `mobilePlansTablesOnSignup`.
PR for AB test: https://github.com/Automattic/wp-calypso/pull/23454

### CODE REVIEW

File: client/jetpack-connect/controller.js [hunk 1]
```diff
@@ -282,7 +282,9 @@ export function plansLanding( context, next ) {
 
 	context.store.dispatch( setTitle( translate( 'Plans', { textOnly: true } ) ) );
 
-	analytics.tracks.recordEvent( 'calypso_plans_view' );
+	analytics.tracks.recordEvent( 'calypso_plans_view', {
+		small_screen: isSmallScreen(),
+	} );
```
Comment (sirreal) [inline client/jetpack-connect/controller.js]: This event is fired anytime you view plans. This is likely the only event you need and you can use different event props for filtering:

* `siteidlabel` `jetpack || wpcom`. Careful, Atomic reports "jetpack"
* `browserdocumentlocation` Track flows you're interested in, like https://wordpress.com/jetpack/conect/plans/SITE_SLUG. Requires fuzzy matching, not sure if Hypothesis will support

If `browserdocumentlocation` won't work in hypothesis, we'll need to find another approach.

**Comment by keoshi** [comment]
@sirreal The test is only valid for signups because we're already rendering the plans in a vertical fashion everywhere else. Supposedly the only two entry points for this test are:

- WPCOM: https://wordpress.com/start
- JP: https://wordpress.com/jetpack/connect/store

But you're right, I might have to specifically target these paths as well, to make sure we're optimizing for these flows and for conversions specifically.

I'll address the rest of the feedback inline.

Thank you!

File: client/my-sites/plan-features/index.jsx [hunk 2]
```diff
@@ -612,8 +612,21 @@ class PlanFeatures extends Component {
...
+		}
+
+		return '0';
+	}
+
 	componentWillMount() {
-		this.props.recordTracksEvent( 'calypso_wp_plans_test_view' );
+		this.props.recordTracksEvent( 'calypso_wp_plans_test_view', {
+			small_screen: this.isSmallScreen(),
+		} );
```
[the remark this answers is not in the document]
Comment (keoshi) [reply client/my-sites/plan-features/index.jsx]: I think we do need this for WPCOM on `/start/`, unless targeting something like `calypso_signup_start`/`calypso_signup_step_start` is better. Those are the only two I can see could carry this property.

File: client/jetpack-connect/controller.js (hunk 1 above)
Comment (keoshi) [reply client/jetpack-connect/controller.js]: How does this sound?

- `calypso_page_view` > `path: /jetpack/connect/store/:site`
- `calypso_plans_view` > `small_screen: 1`
- `wpcom_new_purchase_complete` > `service_slug: automattic`

Comment (sirreal) [reply client/jetpack-connect/controller.js]: > calypso_abtest_start

I don't think you need event at all, hypothesis should take care of things.

Much of this isn't too relevant if dedicated events are added just for this test 🙂

**Comment by sirreal** [comment]
>  The test is only valid for signups

We'll definitely need to target signups better. I like the idea of dedicated, disposable events (one for Jetpack and another for WordPress.com) that can be added for this test and removed later (I saw this was discussed off-GH).

---

> - JP: https://wordpress.com/jetpack/connect/store

The Jetpack connect plans page can be seen here without a connected site, but is also shown during the connection flow at https://wordpress.com/jetpack/connect/plans/SITE_SLUG. Both probably make sense for this test.

**Comment by keoshi** [comment]
Thanks, @sirreal !! I guess I'll just create a new PR with the new events.

**Comment by keoshi** [comment]
Closing in favor of https://github.com/Automattic/wp-calypso/pull/23802

**Pull request closed without merging**